coding-the-legislation/35_periods.md

```diff
@@ -20,15 +62,21 @@ class salary(Variable):
         ...
 ```
 
-The size of the period is constrained by the class attribute `definition_period`:
-  - `definition_period = MONTH`: The variable may have a different value each month. *For example*, the salary of a person. When `formula` is executed, the parameter `period` will always be a whole month. Trying to compute `salary` with a period that is not a month will raise an error before entering `formula`.
+Most of the values calculated in OpenFisca, such as `income tax`, and `housing allowance`, can change over time. 
+
+Therefore, all OpenFisca variables have a `definition_period` attribute.
+The size of the Period is constrained by the class attribute `definition_period`:
+  - `definition_period = MONTH`: The variable may have a different value each month. *For example*, the salary of a person. When `formula` is executed, the parameter `period` will always be a whole month. Trying to compute `salary` with a Period that is not a month will raise an error before entering `formula`.
   - `definition_period = YEAR`: The variable is defined for a year or it has always the same value every months of a year. *For example*, if taxes are to be paid yearly, the corresponding variable is yearly. When `formula` is executed, the parameter `period` will always be a whole year (from January 1st to December 31th).
   - `definition_period = ETERNITY`: The value of the variable is constant. *For example*, the date of birth of a person never changes. `period` is still the 2nd parameter of `formula`. However when `formula` is executed, the parameter `period` can be anything and it should not be used.
 
+Each formula calculates a variable **for the given definition Period**. This Period is always the second argument of the formulas.
+
+## Periods in formulas
 
-## Calculating dependencies for a period different than the one they are defined for
+### Calculate dependencies for a Period different than the variable's `definition_period`
```

Comment:
Reading this page, I got confused between talking about a period of time, and an OpenFisca period.
Capitalizing `period`helps me understand we are talking about an OpenFisca period and not the common name.

Reply:
I can classify  three types of variable in openfisca:
 - flux variable that are linked to some period (= interval of time). Ex: `salaire_imposable`
 - stock variable that are defined to a precise instant (some date). Ex: patrimoine immobilier
 - a "state" (didn't find a better name) variable that are defined to a precise instant (some date). Ex: `boursier` `nombre_enfants`

In some way stock variable can be seen as state variable.

Since we are mainly dealing with period based flux variables, we use stock and state variable over periods using the smallest relevant duration (from the legislation point of view) and assume that the related instant is the beginning of that period. 

But we may carry over a distinction, but fo the time being I do not think it is worthing it.